docs/designs/2730-visibility.md

```diff
@@ -0,0 +1,292 @@
+once the proper logs are found or statuses are checked. The hard part is
+usually understanding what has happened and where to look to find this
+information.
+
+Whether the cluster deployment is successful or fails, it can also be confusing
+to understand what is happening during deployment. This lack of visibility can
+be a challenge to understand where things failed, but it can also be confusing
+during a successful deployment. Especially for new users, this lack of
+visibility can introduce uncertainty that decrease confidence in Tanzu
+Community Edition.
+
+Simplifying the output of our deployment process, even though it provides less
+data along the way, should make it more clear what is happening and help users
+understand where in the deployment process we are. That gives them some
+confidence that things are progressing, and help them understand what is
+happening during that process.
+
+Better output on failure should lead the user to finding root cause and faster
+resolution of blocking issues.
+
+### Goals
+
+* Minimize what gets emitted to the console during cluster deployment to only
+  what the user needs to see
+* Allow more details only if requested when the user requests verbose output
+* Group output into the large grain steps of the process
+* Within each step, only emit the details related to that step that may be
+  useful for the end user to know
+* In case of failure, perform some basic triaging to provide the user with
+  enough details to help them correct the situation
+* Provide clear and direct next steps for them to perform
+
+The hope is to move from the unformatted and sometimes messy output we have
+today:
+
+![current](imgs-2730/current.png)
+
+to a more structured and visually easier to parse format similar to the `kind`
+output:
+
+![kind](imgs-2730/kind.png)
+
+or the planned output for the proposed `unmanaged-cluster` command:
+
+![example](imgs-2730/tce-unmanaged.png)
+
+### Non-Goals/Future Work
+
+This proposal is only focused on cleaning up the output that happens during
+cluster deployment and the information captured on failure. While there may be
+other improvements while this is being implemented, the proposal does not
+include:
+
+* Rearchitecting the means of cluster deployment
+* Auto-remediation of failure cases
+
+We may want to consider updating other command's output such as `tanzu
+management-cluster delete` and `tanzu cluster delete` to match the same style,
+but that is not a primary goal of this proposal.
+
+## Proposal
+
+To implement these changes, the following steps will need to be taken
+
+* Review and group current output into high level steps
+* Review all output and ensure it is at an appropriate log level
+  * Low level details should be at a higher verbosity level so they are not
+    seen unless requested
+  * What is emitted at normal logging levels should be reviewed for its
+    usefulness
+  * All output should be reviewed and removed if it only serves as development
```

Comment:
I'd prefer these get pushed down to a `debug` level rather than outright removed. If they're gone entirely, it can make for more needle-in-haystack problems for those who _do_ want to get to a code path level of detail.

I'm all for making sure they require the highest level of verbosity, though. Most of the time users do not need them.

Reply:
Yeah, that makes sense.